src/Tutor.Infrastructure/Serialization/EventSerializer.cs

```diff
@@ -5,14 +5,16 @@
 using System.Text.Json;
 using Tutor.Core.BuildingBlocks.EventSourcing;
 using Tutor.Core.DomainModel.KnowledgeComponents;
+using Tutor.Core.DomainModel.KnowledgeComponents.AssessmentEventHelp;
 
 namespace Tutor.Infrastructure.Serialization
 {
     public static class EventSerializer
     {
         private static readonly IDictionary<Type, string> eventRelatedTypes = new Dictionary<Type, string>()
         {
-            { typeof(AssessmentEventAnswered), "AssessmentEventAnswered"}
+            { typeof(AssessmentEventAnswered), "AssessmentEventAnswered" },
+            { typeof(SoughtHelp), "SoughtHelp" }
```

Comment:
Maybe we should separate "SoughtHints" from "ViewedSolution"?

Reply:
Separated hints and solution into classes in [b2375d3](https://github.com/Clean-CaDET/tutor/pull/26/commits/b2375d3acc0bada1e63dc0824c47dade4d026d31). That's actually the design I initially had for this, but I switched to the version with HelpType because it seemed simpler and I didn't really see any big advantages in either approach at this point, since the general goal is really just to record the events and the functionality itself isn't particularly developed yet.

Reply:
SoughtChallengeSolution sounds good. I'm not really sure about the ancestor, as viewing the solution essentially transforms the AE (challenge) into an IE (video).

Reply:
From that perspective I definitely agree that it would be good to reconsider them having this exact common ancestor, but I think there's no harm in leaving it as is for now (it achieves the main goal of storing the events, and there isn't any information about ancestors in the stored version anyway). 
I feel like the main issue with designing this right now is that really, other than just storing the events, we have no idea what exactly (if anything) we want to do with the exact information that the user viewed hints or the solution of a challenge. The best solution I see is to make KcMastery have a functionality that just receives events from the outside and stores them, but obviously there has to be some sort of validation for which events to accept. Then we can add any events we want to that infrastructure, and just extract them from there into their own methods if we ever want to add an actual functionality related to those events. But I'm honestly really not sure how to design that to make sure that it's just enough of a free slate, but not too much to a point where it can accept just anything. That's why I went with a skeleton of a 'seek help' functionality for these two events instead, but I would really like to discuss better ways to deal with this on our next sync.

Reply:
One attempt at an alternate design for this is in https://github.com/Clean-CaDET/tutor/pull/29.